backend/src/main/java/de/adrianaschepers/shishabuddies/controller/UserController.java

```diff
@@ -63,17 +64,43 @@ public UserController(UserService userService) {
         return ok(createdSettings);
     }
 
+    @PutMapping("update-settings")
+    public ResponseEntity<Settings> updateSettings(@RequestBody Settings newSettings, @AuthenticationPrincipal UserEntity authUser){
+        SettingsEntity newSettingsEntity = map(newSettings); //new settigns
+        SettingsEntity updateSettingsEntity = userService.getUserSettings(authUser); //get old settings of user
+
+       if(!updateSettingsEntity.getFavHookah().equals(newSettingsEntity.getFavHookah())){
+           updateSettingsEntity.setFavHookah(newSettingsEntity.getFavHookah());
+       }
+       if(!updateSettingsEntity.getFavHookahHead().equals(newSettingsEntity.getFavHookahHead())){
+           updateSettingsEntity.setFavHookahHead(newSettingsEntity.getFavHookahHead());
+       }
+       if(!updateSettingsEntity.getFavTobacco().equals(newSettingsEntity.getFavTobacco())){
+           updateSettingsEntity.setFavTobacco(newSettingsEntity.getFavTobacco());
+       }
+       if(updateSettingsEntity.getNumberOfHookahs()!= newSettingsEntity.getNumberOfHookahs()){
+           updateSettingsEntity.setNumberOfHookahs(newSettingsEntity.getNumberOfHookahs());
+       }
+       if(updateSettingsEntity.getNumberOfHookahHeads()!= newSettingsEntity.getNumberOfHookahHeads()){
+           updateSettingsEntity.setNumberOfHookahHeads(newSettingsEntity.getNumberOfHookahHeads());
+       }
+       if(updateSettingsEntity.getNumberOfTobaccos()!= newSettingsEntity.getNumberOfTobaccos()){
+           updateSettingsEntity.setNumberOfTobaccos(newSettingsEntity.getNumberOfTobaccos());
+       }
+       SettingsEntity updatedSettingsEnt= userService.createSettings(updateSettingsEntity,authUser);
```

Comment:
das du in du im update endpunkt die createSettings methode verwendest finde ich verwirrend. nenne sie vielleicht einfach saveSettings

Reply:
danke!